packages/flutter/test/widgets/editable_text_test.dart

```diff
@@ -16154,6 +16154,35 @@ void main() {
       skip: !kIsWeb, // [intended]
     );
 
+    testWidgets('when setCustomSelectionOnFocus is set', (WidgetTester tester) async {
+      // Regression test for https://github.com/flutter/flutter/issues/163399.
+      controller.text = 'Text';
+
+      await tester.pumpWidget(
+        MaterialApp(
+          home: EditableText(
+            key: ValueKey<String>(controller.text),
+            controller: controller,
+            focusNode: focusNode,
+            style: Typography.material2018().black.titleMedium!,
+            cursorColor: Colors.blue,
+            backgroundCursorColor: Colors.grey,
+            setCustomSelectionOnFocus: () => controller.selection,
```

Comment:
Nit: Maybe set this to something different so that it's more obvious that it worked, and not that it just kept the existing selection.